packages/credential-provider-node/src/defaultProvider.ts

```diff
@@ -58,14 +63,36 @@ export type DefaultProviderInit = FromIniInit &
 export const defaultProvider = (init: DefaultProviderInit = {}): MemoizedProvider<AwsCredentialIdentity> =>
   memoize(
     chain(
-      ...(init.profile || process.env[ENV_PROFILE]
-        ? []
-        : [
-            async () => {
-              init.logger?.debug("@aws-sdk/credential-provider-node - defaultProvider::fromEnv");
-              return fromEnv(init)();
-            },
-          ]),
+      async () => {
+        const profile = init.profile ?? process.env[ENV_PROFILE];
+        if (profile) {
+          const envStaticCredentialsAreSet = process.env[ENV_KEY] && process.env[ENV_SECRET];
+          if (envStaticCredentialsAreSet) {
+            if (!multipleCredentialSourceWarningEmitted) {
+              const warnFn =
+                init.logger?.warn && init.logger?.constructor?.name !== "NoOpLogger" ? init.logger.warn : console.warn;
+              warnFn(
```

Comment:
This warning is emitted here in the default provider chain rather than in the `fromEnv` provider function.

The reason for this is because when using the `fromEnv` provider in isolation, it doesn't matter if the `AWS_PROFILE` is set or not.